test/github.ts

```diff
@@ -809,6 +811,61 @@ describe('GitHub', () => {
       expect(release!.tag_name).to.eql('v1.2.3');
       expect(release!.draft).to.be.false;
     });
+
+    it('should raise a DuplicateReleaseError if already_exists', async () => {
+      req
+        .post('/repos/fake/fake/releases', body => {
+          snapshot(body);
+          return true;
+        })
+        .reply(422, {
+          message: 'Validation Failed',
+          errors: [
+            {
+              resource: 'Release',
+              code: 'already_exists',
+              field: 'tag_name',
+            },
+          ],
+          documentation_url:
+            'https://docs.github.com/rest/reference/repos#create-a-release',
+        });
+
+      const promise = github.createRelease(
+        '',
+        'v1.2.3',
+        'abc123',
+        'Some release notes',
+        false
+      );
+      await assert.rejects(promise, error => {
+        return error instanceof DuplicateReleaseError;
+      });
+    });
+
+    it('should raise a RequestError for other validation errors', async () => {
```

Comment:
I would add a unit test that confirms the original stack trace information wasn't lost.

Reply:
The assertion now ensures that the calling method in the GitHub class is in the stacktrace. In this case, `GitHub.createRelease`.